solr/solrj/src/java/org/apache/solr/client/solrj/impl/BaseHttpClusterStateProvider.java

```diff
@@ -182,6 +157,37 @@ private ClusterState fetchClusterState(
     return cs;
   }
 
+  private DocCollection getDocCollectionFromObjects(Map.Entry<String, Object> e, int zNodeVersion) {
+    @SuppressWarnings("rawtypes")
+    Map m = (Map) e.getValue();
+    Long creationTimeMillisFromClusterStatus = (Long) m.get("creationTimeMillis");
+    Instant creationTime =
+        creationTimeMillisFromClusterStatus == null
+            ? Instant.EPOCH
+            : Instant.ofEpochMilli(creationTimeMillisFromClusterStatus);
+    return fillPrs(zNodeVersion, e, creationTime, m);
```

Comment:
ah, you chose Map.Entry because you want to pass it to `fillPrs`, which already exists and takes this.  I'd update that method as well instead.